aioaws/s3.py

```diff
@@ -173,8 +173,10 @@ def signed_upload_url(
         assert path == '' or path.endswith('/'), 'path must be empty or end with "/"'
         assert not path.startswith('/'), 'path must not start with "/"'
         key = path + filename
+        bucket = get_config_attr(self._config, 'aws_s3_bucket')
+        bucket_name = bucket.split(sep='.')[0] if '.' in bucket else bucket
```

Comment:
That's right, using `.` in bucket names is a common practice for serving static content. AWS has been trying to deprecate `.` in bucket names for a while (as part of this whole "[virtual-hosted-style URL](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/)" thing), but it's still allowed at this time.

I'm thinking:

- Keep the same behavior for now, but add some docs explaining it with this PR
- Implement the suggested regex in a separate PR, and use it consistently throughout the package. I'm happy to take this on. This would close #22, but would introduce other limitations. For example, `aioaws` could only be used on AWS S3 specifically, and not other S3-compatible platforms, like Backblaze B2, Cloudflare R2 (soon), DigitalOcean Spaces, Linode Object Storage, etc. Another option could be to simply add another config field for the bucket name, and not attempt to parse the bucket name out of the hostname.

How does that sound?

Reply:
Sorry for the slow reply, let's go with the regex solution.

Reply:
> Sorry for the slow reply, let's go with the regex solution.

Sounds good. Updated to use your suggested regex. Tests updated accordingly.